### ISSUE TITLE
The Great Stem Debate of Assignment 3

### ISSUE THREAD

**Issue by dbajic** [iit-cs429/main#55]
A couple of issues/queries/concerns/whatever:

You mentioned the basis for assignment 3 was assignment 2, with tweaking it to add RSV, BM25, etc. on it, and that the stemming/cosine similarity functions we've already written should work. However I noticed right away in trying some of the queries given in the TIME dataset that it breaks spectacularly. 

It seems that some of the terms in the query are stemmed in a way that is not covered by our current stemmer (Example: query 1 uses `suppressing`, however `suppressing` appears nowhere in TIME.ALL, only `suppress` and `suppressed`. Similarly for query 2, the query uses the term `repression`, but only `repressions` appears in any of the documents, and hence `repression` won't appear in the index.)

The other issue that I felt would happen that I brought up in class (that happened) was that `U.S.` would not appear in the index, which kind of throws everything in a loop because there might be other terms like this in the queries that may not appear in the index, and might cause a bunch of headaches.

tl;dr: Stemming isn't doing what I was expecting and now I'm a bit confused because there's multiple, different cases that cause searching to break. Not sure what I should/could do in this case.

**Comment by aronwc** [comment]
- If a term appears in the query but not in the index, then you may ignore this term when ranking (e.g., in query 1, "suppressing" would not influence RSV scores). This situation can happen even if you do more complex stemming (e.g., I could enter a search term "asdfghjkl")
- U.S. should be tokenized into the terms "U" and "S" in both the query and the document, so this should still work (although it will obviously be imperfect, as it would match documents with abbreviations like "U.S.D.A.").

I'm not sure exactly what you mean "cases that cause searching to break." Do you mean it is returning the wrong value, or it is crashing?

**Comment by dbajic** [comment]
If I were to search for that case then the search would fail because of a division error, but then I remembered that I also have to check if it's in the index, which SHOULD fix the issue. I'll let you know when I get the chance.

**Comment by dbajic** [comment]
Yep, that did the trick and fixed my issues.
